ggshield/pre_receive_cmd.py

```diff
@@ -0,0 +1,149 @@
+    """Test all options passed to git for `breakglass`"""
+    raw_option_count = os.getenv("GIT_PUSH_OPTION_COUNT", None)
+    if raw_option_count is not None:
+        option_count = int(raw_option_count)
+        for option in range(option_count):
+            if os.getenv(f"GIT_PUSH_OPTION_{option}", "") == "breakglass":
+                return True
+
+    return False
+
+
+@click.command()
+@click.argument("prereceive_args", nargs=-1, type=click.UNPROCESSED)
+@click.pass_context
+@exit_after(PRERECEIVE_TIMEOUT)
+def prereceive_cmd(ctx: click.Context, prereceive_args: List[str]) -> int:
+    """
+    scan as a pre-push git hook.
+    """
+    config = ctx.obj["config"]
+
+    breakglass = get_breakglass_option()
+    if breakglass:
+        click.echo("SKIP: breakglass detected. Skipping GitGuardian pre-receive hook.")
+        return 0
+
+    args = sys.stdin.read().strip().split()
+    if len(args) < 3:
+        raise click.ClickException(f"Invalid input arguments: {args}")
+
+    oldref, newref, *_ = args
+
+    if newref == EMPTY_SHA:
+        click.echo("Deletion event or nothing to scan.")
+        return 0
+
+    if oldref == EMPTY_SHA:
+        click.echo(
+            f"New tree event. Scanning last {config.max_commits_for_hook} commits."
+        )
+        before = EMPTY_TREE
+        after = newref
+        cmd_range = f"--max-count={config.max_commits_for_hook+1} {EMPTY_TREE} {after}"
+    else:
+        before = oldref
+        after = newref
+        cmd_range = (
+            f"--max-count={config.max_commits_for_hook+1} {before}...{after}"  # noqa
+        )
+
+    commit_list = get_list_commit_SHA(cmd_range)
+
+    if not commit_list:
+        click.echo(
+            "Unable to get commit range.\n"
+            f"  before: {before}\n"
+            f"  after: {after}\n"
+            "Skipping pre-receive hook\n"
+        )
+        return 0
+
+    if len(commit_list) > config.max_commits_for_hook:
+        click.echo(
+            f"Too many commits. Scanning last {config.max_commits_for_hook} commits\n"
+        )
+        commit_list = commit_list[-config.max_commits_for_hook :]
+
+    if config.verbose:
+        click.echo(f"Commits to scan: {len(commit_list)}")
+
+    try:
+        check_git_dir()
+        return_code = scan_commit_range(
+            client=ctx.obj["client"],
+            cache=ctx.obj["cache"],
+            commit_list=commit_list,
+            output_handler=ctx.obj["output_handler"],
+            verbose=config.verbose,
+            filter_set=ctx.obj["filter_set"],
+            matches_ignore=config.matches_ignore,
+            all_policies=config.all_policies,
+            scan_id=" ".join(commit_list),
+            mode_header=SupportedScanMode.PRE_RECEIVE.value,
+            banlisted_detectors=config.banlisted_detectors,
+        )
+        if return_code:
+            click.echo(
+                """Rewrite your git history to delete evidence of your secrets.
```

Comment:
pluralize `secret` maybe

Reply:
Not enough context for this, will skip for now as discussed